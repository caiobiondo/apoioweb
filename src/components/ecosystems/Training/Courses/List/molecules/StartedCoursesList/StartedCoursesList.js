import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { translate } from 'locale';
import Slider from 'react-slick';
import PropTypes from 'prop-types';
import {
  Header,
  Wrapper,
  CourseInCoverInfo,
  CourseCategoryTitle,
  CourseTitle,
  LeftCarouselArrow,
  RightCarouselArrow,
  StartedCoursesWrapper,
  StartedCoursesThumbnail,
} from './StartedCoursesList.styles';
import {
  TrainingCoursesQuery,
  TrainingCoursesQueryOptions,
} from 'components/ecosystems/Training/data/TrainingCourses.data';
import { graphql } from 'react-apollo';

import ImageWithFallback from 'components/molecules/ImageWithFallback';

export class StartedCoursesList extends Component {
  handleCourseClick = course => event => {
    const url = course.type === 'VIDEO' ? `${course.id}` : `${course.id}/start`;

    this.props.history.push(`/training/courses/${url}`);
  };

  render() {
    const settings = {
      arrows: true,
      autoplay: false,
      autoplaySpeed: 5000,
      adaptiveHeight: true,
      dots: true,
      draggable: true,
      infinite: false,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      prevArrow: <LeftCarouselArrow>{'<'}</LeftCarouselArrow>,
      nextArrow: <RightCarouselArrow>{'>'}</RightCarouselArrow>,
      responsive: [
        {
          breakpoint: 768,
          settings: {
            dots: false,
            draggable: true,
            autoplay: false,
          },
        },
      ],
    };
    const { courses } = this.props;
    if (!courses) return null;
    return (
      <Wrapper>
        <Header>{translate('continueCourses')}</Header>
        <StartedCoursesWrapper>
          <Slider {...settings}>
            {courses.map(course => {
              return (
                <StartedCoursesThumbnail key={course.id}>
                  <CourseInCoverInfo onClick={this.handleCourseClick(course)}>
                    <CourseTitle>{course.title}</CourseTitle>
                    <CourseCategoryTitle>{course.categoryTitle}</CourseCategoryTitle>
                  </CourseInCoverInfo>
                  <ImageWithFallback imageUrl={course.thumbnail} />
                </StartedCoursesThumbnail>
              );
            })}
          </Slider>
        </StartedCoursesWrapper>
      </Wrapper>
    );
  }
}

StartedCoursesList.propTypes = {
  user: PropTypes.object.isRequired,
  status: PropTypes.string.isRequired,
};

export const StartedCoursesListWithRouter = withRouter(StartedCoursesList);

export default graphql(TrainingCoursesQuery, TrainingCoursesQueryOptions)(
  StartedCoursesListWithRouter,
);
