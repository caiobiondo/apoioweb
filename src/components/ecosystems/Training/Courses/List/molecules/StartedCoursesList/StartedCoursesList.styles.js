import styled from 'styled-components';
import { Responsive, getTheme } from '@entria/components';
import { black, white } from 'styles/colors';
import { RobotoMedium, NaturaRegular } from 'styles/typography';

export const Wrapper = styled.div`
  margin-top: 52px;
  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    position: relative;
    padding: 0px 10px 0px 10px;
  }
`;

export const Header = styled.h2`
  font-family: ${NaturaRegular};
  font-size: 30px;
  font-weight: bold;
  line-height: 0.4;
  text-align: left;
  color: ${black};
  margin-bottom: 28px;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 25px;
    padding: 20px;
    margin-bottom: -22px;
  }
`;

export const StartedCoursesWrapper = styled.div`
  position: relative;
  height: 400px;
  width: 100%;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    height: 240px;
  }

  > div {
    position: absolute;
    width: 100%;
    overflow-x: hidden;
  }

  .slick-list {
    height: 100% !important;
  }

  .slick-slide {
    display: inline-block;
  }

  .slick-dots {
    position: absolute;
    bottom: 20px;
    display: flex !important;
    left: 0px;
    right: 0px;
    align-items: center;
    justify-content: center;

    list-style: none;

    li {
      position: relative;
      display: inline-block;
      width: 20px;
      height: 20px;
      margin: 0 5px;
      padding: 0;
      cursor: pointer;

      button {
        font-size: 0;
        line-height: 0;
        display: block;
        width: 20px;
        height: 20px;
        padding: 5px;
        cursor: pointer;
        color: transparent;
        border: 0;
        outline: none;
        background: transparent;

        &:before {
          position: absolute;
          content: '';
          width: 10px;
          height: 10px;
          border-radius: 50%;
          background-color: white;
          border: 1px solid #f3971f;
        }
      }

      &.slick-active button:before {
        background-color: #f3971f;
      }
    }
  }
`;

export const CourseInCoverInfo = styled.div`
  position: absolute;
  top: 0;
  bottom: 5px;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.5);
  padding: 30px;
  cursor: pointer;
`;

export const CourseTitle = styled.h2`
  font-family: ${NaturaRegular};
  font-size: 38.8px;
  font-weight: bold;
  line-height: 1.25;
  text-align: left;
  color: ${white};
  margin: -87px 0px 40px 85px;
  text-shadow: 0px 2.5px 15px rgba(0, 0, 0, 0.1);
  position: relative;

  &:after {
    bottom: -10px;
    left: 0;
    width: 40px;
    height: 2px;
    background: ${getTheme().palette.primary1Color};
    content: '';
    position: absolute;
  }

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin: -30px 0px 40px 47px
    font-size: 20px;
  }

  @media (max-width: 481px) {
    margin: -10px 0px 40px 47px;
    font-size: 20px;
  }
`;

export const CourseCategoryTitle = styled.span`
  font-family: ${RobotoMedium};
  font-size: 20.5px;
  font-weight: normal;
  line-height: 0.71;
  text-align: left;
  text-transform: uppercase;
  color: ${white};
  margin-left: 84px;
  text-shadow: 0px 2.5px 15px rgba(0, 0, 0, 0.1);

  @media (max-width: 481px) {
    margin-left: 46px;
    font-size: 10px;
  }
`;

export const StartedCoursesThumbnail = styled.div`
  display: flex;
  justify-content: center;
  position: relative;
  margin-top: 28px;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    height: 350px
  }

    img {
      width: 100%;
      object-fit: cover;
      height: 350px;

      @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
        height: 195px
      }
    }

  }
`;

export const LeftCarouselArrow = styled.span`
  position: absolute;
  font-size: 100px;
  color: #fff;
  display: flex;
  align-items: center;
  z-index: 2;
  top: 38%;
  text-shadow: 1px 0px 4px black;
  cursor: pointer;
  margin-left: 23px;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 70px;
    top: 38%;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 70px;
    top: 38%;
  }
`;

export const RightCarouselArrow = LeftCarouselArrow.extend`
  right: 23px;
`;
