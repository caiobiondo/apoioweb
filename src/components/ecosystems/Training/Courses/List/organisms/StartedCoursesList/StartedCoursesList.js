import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { translate } from 'locale';
import Slider from 'react-slick';
import PropTypes from 'prop-types';
import {
  Header,
  Wrapper,
  StartedCourseInfo,
  StartedCourseCategoryTitle,
  StartedCourseTitle,
  LeftCarouselArrow,
  RightCarouselArrow,
  StartedCoursesWrapper,
  StartedCoursesThumbnail,
} from './StartedCoursesList.styles';
import {
  TrainingCoursesQuery,
  TrainingCoursesQueryOptions,
} from 'components/ecosystems/Training/data/TrainingCourses.data';
import { graphql } from 'react-apollo';

import ImageWithFallback from 'components/molecules/ImageWithFallback';

class LeftArrow extends Component {
  render() {
    const { onClick, currentSlide } = this.props;
    if (currentSlide === 0) {
      return null;
    }
    return <LeftCarouselArrow onClick={onClick}>{'<'}</LeftCarouselArrow>;
  }
}

class RightArrow extends Component {
  render() {
    const { onClick, currentSlide, slideCount, slidesToScroll } = this.props;
    if (currentSlide + slidesToScroll >= slideCount) {
      return null;
    }
    return <RightCarouselArrow onClick={onClick}>{'>'}</RightCarouselArrow>;
  }
}

export class StartedCoursesList extends Component {
  handleCourseClick = course => event => {
    const url = course.type === 'VIDEO' ? `${course.id}` : `${course.id}/start`;

    this.props.history.push(`/training/courses/${url}`);
  };
  render() {
    const settings = {
      arrows: true,
      autoplay: false,
      autoplaySpeed: 5000,
      adaptiveHeight: true,
      dots: true,
      draggable: true,
      infinite: false,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      responsive: [
        {
          breakpoint: 768,
          settings: {
            dots: false,
            draggable: true,
            autoplay: false,
          },
        },
      ],
      prevArrow: <LeftArrow />,
      nextArrow: <RightArrow slidesToScroll={1} />,
    };
    const { courses } = this.props;
    if (!courses) return null;
    return (
      <Wrapper>
        <Header>{translate('continueCourses')}</Header>
        <StartedCoursesWrapper>
          <Slider {...settings}>
            {courses.map(course => {
              return (
                <StartedCoursesThumbnail key={course.id}>
                  <StartedCourseInfo onClick={this.handleCourseClick(course)}>
                    <StartedCourseTitle>{course.title}</StartedCourseTitle>
                    <StartedCourseCategoryTitle>{course.categoryTitle}</StartedCourseCategoryTitle>
                  </StartedCourseInfo>
                  <ImageWithFallback imageUrl={course.thumbnail} />
                </StartedCoursesThumbnail>
              );
            })}
          </Slider>
        </StartedCoursesWrapper>
      </Wrapper>
    );
  }
}

StartedCoursesList.propTypes = {
  user: PropTypes.object.isRequired,
  status: PropTypes.string.isRequired,
};

export const StartedCoursesListWithRouter = withRouter(StartedCoursesList);

export default graphql(TrainingCoursesQuery, TrainingCoursesQueryOptions)(
  StartedCoursesListWithRouter,
);
