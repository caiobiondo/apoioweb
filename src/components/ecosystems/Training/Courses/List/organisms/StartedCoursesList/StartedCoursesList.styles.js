import styled from 'styled-components';
import { Responsive, getTheme } from '@entria/components';
import { black, white, orange100, gray100 } from 'styles/colors';
import {
  RobotoMedium,
  NaturaRegular,
  fs15,
  fs20,
  fs25,
  fs30,
  fs60,
  fs100,
} from 'styles/typography';

export const Wrapper = styled.div`
  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    position: relative;
    padding: 0px 10px 0px 10px;
  }

  margin-bottom: 24px;
`;

export const Header = styled.h2`
  font-family: ${NaturaRegular};
  font-size: ${fs30};
  font-weight: bold;
  text-align: left;
  color: ${black};
  padding-top: 15px;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: ${fs25};
    padding: 0 0 20px 0;
    margin: 0;
  }
`;

export const StartedCoursesWrapper = styled.div`
  position: relative;
  height: 355px;
  width: 100%;

  @media (max-height: ${Responsive.VIEWPORT.MEDIUM}px) and (max-width: 975px), (max-width: 1024px) {
    height: 195px;
  }

  .slick-slider {
    position: absolute;
    width: 100%;
    overflow: hidden;
    height: 100%;
  }

  .slick-list {
    height: 100% !important;
  }

  .slick-track {
    height: 100%;
  }

  .slick-slide {
    display: inline-block;
  }

  .slick-dots {
    position: absolute;
    bottom: 20px;
    display: flex !important;
    left: 0px;
    right: 0px;
    align-items: center;
    justify-content: center;

    list-style: none;

    li {
      position: relative;
      display: inline-block;
      width: 20px;
      height: 20px;
      margin: 0 5px;
      padding: 0;
      cursor: pointer;

      button {
        font-size: 0;
        line-height: 0;
        display: block;
        width: 20px;
        height: 20px;
        padding: 5px;
        cursor: pointer;
        color: transparent;
        border: 0;
        outline: none;
        background: transparent;

        &:before {
          position: absolute;
          content: '';
          width: 10px;
          height: 10px;
          border-radius: 50%;
          background-color: white;
          border: 1px solid ${orange100};
        }
      }

      &.slick-active button:before {
        background-color: ${orange100};
      }
    }
  }
`;

export const StartedCourseInfo = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  padding: 30px;
  cursor: pointer;
  z-index: 1;
`;

export const StartedCourseTitle = styled.h2`
  font-family: ${NaturaRegular};
  font-size: 38.8px;
  font-weight: bold;
  line-height: 1.25;
  text-align: left;
  color: ${white};
  margin: -87px 0px 40px 85px;
  text-shadow: 0px 2.5px 15px rgba(0, 0, 0, 0.1);
  position: relative;

  &:after {
    bottom: -10px;
    left: 0;
    width: 40px;
    height: 2px;
    background: ${getTheme().palette.primary1Color};
    content: '';
    position: absolute;
  }

  @media (max-width: 1024px) {
    margin: -15px 0px 40px 47px;
    font-size: ${fs20};
  }
`;

export const StartedCourseCategoryTitle = styled.span`
  font-family: ${RobotoMedium};
  font-size: 20.5px;
  font-weight: normal;
  text-align: left;
  text-transform: uppercase;
  color: ${white};
  margin-left: 84px;
  text-shadow: 0px 2.5px 15px rgba(0, 0, 0, 0.1);

  @media (max-width: 1024px) {
    margin-left: 46px;
    font-size: ${fs15};
  }
`;

/* eslint-disable no-confusing-arrow */
export const StartedCoursesThumbnail = styled.div`
  position: relative;
  background-color: ${gray100};
  background-image: url(${props => (props.imageUrl ? props.imageUrl : '')});
  background-repeat: no-repeat;
  background-size: cover;
  height: 100%;

  &:before {
    position: absolute;
    content: '';
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background: rgba(0, 0, 0, 0.5);
    z-index: 1;
  }
`;
/* eslint-enable no-confusing-arrow */

export const LeftCarouselArrow = styled.span`
  position: absolute;
  font-size: ${fs100};
  color: #fff;
  display: flex;
  align-items: center;
  z-index: 2;
  top: 50%;
  bottom: 50%;
  text-shadow: 1px 0px 4px black;
  cursor: pointer;
  margin-left: 23px;

  @media (max-width: 1024px) {
    font-size: ${fs60};
  }
`;

export const RightCarouselArrow = LeftCarouselArrow.extend`
  right: 23px;
`;
