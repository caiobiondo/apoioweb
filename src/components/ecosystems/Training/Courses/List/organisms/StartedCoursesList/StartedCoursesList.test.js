import React from 'react';
import { shallow } from 'enzyme';
import { LeftArrow, RightArrow, StartedCoursesList } from './StartedCoursesList';

describe('StartedCourses', () => {
  it('should call StartedCourse and redirect to a new router', () => {
    const props = {
      history: {
        push: jest.fn(),
      },
      status: 'started',
      courses: [
        {
          id: '1',
          title: 'Como entender as necessidades dos clientes',
          type: 'VIDEO',
          categoryTitle: '',
          thumbnail: '',
        },
      ],
    };

    const result = shallow(<StartedCoursesList {...props} />);
    const instance = result.instance();
    instance.handleCourseClick(props.courses[0])({});

    expect(props.history.push).toBeCalledWith(`/training/courses/1/video`);
  });

  it('should call StartedCourse and redirect to a new router and not a VIDEO', () => {
    const props = {
      history: {
        push: jest.fn(),
      },
      status: 'started',
      courses: [
        {
          id: '1',
          title: 'Como entender as necessidades dos clientes',
          type: 'HTML5',
          categoryTitle: '',
          thumbnail: '',
        },
      ],
    };

    const result = shallow(<StartedCoursesList {...props} />);
    const instance = result.instance();
    instance.handleCourseClick(props.courses[0])({});

    expect(props.history.push).toBeCalledWith(`/training/courses/1/html5`);
  });

  it('should render a slide with the courses', () => {
    // given
    const props = {
      status: 'started',
      courses: [
        {
          id: 1,
          title: 'Como entender as necessidades dos clientes',
          type: 'VIDEO',
          categoryTitle: '',
          thumbnail: '',
        },
      ],
    };

    // when
    const result = shallow(<StartedCoursesList {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should not render a slide when courses are null', () => {
    // given
    const props = {
      courses: null,
      status: 'started',
    };

    // when
    const result = shallow(<StartedCoursesList {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });
});

describe('LeftArrow', () => {
  it('should render null', () => {
    // given
    const props = {
      onClick: () => {},
      currentSlide: 0,
    };

    // when
    const result = shallow(<LeftArrow {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should render a left arrow', () => {
    // given
    const props = {
      onClick: () => {},
      currentSlide: 1,
    };

    // when
    const result = shallow(<LeftArrow {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });
});

describe('RightArrow', () => {
  it('when right arrow is null', () => {
    // given
    const props = {
      onClick: () => {},
      currentSlide: 0,
      slideCount: 0,
      slidesToScroll: 0,
    };
    // when
    const result = shallow(<RightArrow {...props} />);
    // then
    expect(result).toMatchSnapshot();
  });
  it('should render a right arrow', () => {
    // given
    const props = {
      onClick: () => {},
      currentSlide: 2,
      slideCount: 5,
      slidesToScroll: 2,
    };

    // when
    const result = shallow(<RightArrow {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });
});
