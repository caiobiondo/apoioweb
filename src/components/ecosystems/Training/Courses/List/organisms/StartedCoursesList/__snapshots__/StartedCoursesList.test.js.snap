// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftArrow should render a left arrow 1`] = `
<StartedCoursesListstyles__LeftCarouselArrow
  onClick={[Function]}
>
  &lt;
</StartedCoursesListstyles__LeftCarouselArrow>
`;

exports[`LeftArrow should render null 1`] = `""`;

exports[`RightArrow should render a right arrow 1`] = `
<StartedCoursesListstyles__LeftCarouselArrow
  onClick={[Function]}
>
  &gt;
</StartedCoursesListstyles__LeftCarouselArrow>
`;

exports[`RightArrow when right arrow is null 1`] = `""`;

exports[`StartedCourses should not render a slide when courses are null 1`] = `""`;

exports[`StartedCourses should render a slide with the courses 1`] = `
<StartedCoursesListstyles__Wrapper>
  <StartedCoursesListstyles__Header>
    Continue seus cursos
  </StartedCoursesListstyles__Header>
  <StartedCoursesListstyles__StartedCoursesWrapper>
    <Slider
      adaptiveHeight={true}
      arrows={true}
      autoplay={false}
      autoplaySpeed={5000}
      dots={true}
      draggable={true}
      infinite={false}
      nextArrow={
        <RightArrow
          slidesToScroll={1}
        />
      }
      prevArrow={<LeftArrow />}
      responsive={
        Array [
          Object {
            "breakpoint": 768,
            "settings": Object {
              "autoplay": false,
              "dots": false,
              "draggable": true,
            },
          },
        ]
      }
      slidesToScroll={1}
      slidesToShow={1}
      speed={500}
    >
      <StartedCoursesListstyles__StartedCoursesThumbnail
        key="1"
      >
        <StartedCoursesListstyles__StartedCourseInfo
          onClick={[Function]}
        >
          <StartedCoursesListstyles__StartedCourseTitle>
            Como entender as necessidades dos clientes
          </StartedCoursesListstyles__StartedCourseTitle>
          <StartedCoursesListstyles__StartedCourseCategoryTitle />
        </StartedCoursesListstyles__StartedCourseInfo>
        <ImageWithFallback
          fallbackIcon="ico_pictureless"
          imageUrl=""
        />
      </StartedCoursesListstyles__StartedCoursesThumbnail>
    </Slider>
  </StartedCoursesListstyles__StartedCoursesWrapper>
</StartedCoursesListstyles__Wrapper>
`;
